### PR TITLE
Added backwards/gradient tests to the default convolution unit tests.

### PR DESCRIPTION
**Pull request by blackencino**
Backwards/Gradient tests for default backend convolution unit tests.

New tests:

* test_single_impulse_backward

### CODE REVIEW

File: tests/unit/test_conv_default.py [hunk 1]
```diff
@@ -444,3 +444,157 @@ def sort_coords(coords):
...
+        # Anti-symmetric kernel with requires_grad
+        kernel = fourier_anti_symmetric_kernel(self.KERNEL_SIZE, dtype=dtype, device=device)
+        self.assertFalse(has_any_symmetry(kernel))
+        kernel_with_channels = kernel.reshape(1, 1, *self.KERNEL_SIZE).clone()
+        kernel_with_channels.requires_grad_(True)
+        print("Created anti-symmetric kernel with requires_grad=True.")
+
+        # Create convolution plan
+        conv_plan = ConvolutionPlan.from_grid_batch(
+            kernel_size=self.KERNEL_SIZE, stride=1, source_grid=grid_batch, target_grid=dst_grid_batch
```
Comment (swahtz) [inline tests/unit/test_conv_default.py:502]: **❓ question:** ‏Have we not implemented the stride functionality yet?  Should we test over values of this too?

Comment (blackencino) [reply tests/unit/test_conv_default.py:502]: We have not validated the stride functionality yet. It is next on the validation list. I'm adding these tests starting at the ground up, doing them one at a time.

Comment (blackencino) [reply tests/unit/test_conv_default.py:502]: So yes - we should absolutely be testing against striding, but that should be a test that builds on top of these, not woven into these.

Comment (swahtz) [reply tests/unit/test_conv_default.py:502]: OK, great, thanks, just was curious.
